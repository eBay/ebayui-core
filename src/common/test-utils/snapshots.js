import { render, prettyDOM } from "@marko/testing-library";
import { expect } from "vitest";

function snapshotHTML() {
    return async (template, input) => {
        const { container } = await render(template, input);
        expect(prettyDOM(container)).toMatchSnapshot();
    };
}

export { snapshotHTML };
