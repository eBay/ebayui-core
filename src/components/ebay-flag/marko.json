{
    "taglib-id": "ebayui-core-flags",
    "tags-dir": "./icons"
}
