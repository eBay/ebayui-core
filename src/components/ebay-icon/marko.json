{
  "taglib-id": "ebayui-core-icons",
  "tags-dir": "./icons"
}