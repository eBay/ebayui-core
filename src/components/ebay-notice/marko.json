{
    "<ebay-notice>": {
        "renderer": "./index.js"
    }
}
