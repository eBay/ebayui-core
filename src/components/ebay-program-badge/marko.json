{
  "taglib-id": "ebayui-core-program-badges",
  "tags-dir": "./icons"
}
