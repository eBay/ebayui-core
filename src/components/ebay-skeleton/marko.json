{
    "taglib-id": "ebayui-core-skeleton",
    "tags-dir": "./components"
}
