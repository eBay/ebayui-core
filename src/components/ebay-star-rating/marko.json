{
    "taglib-id": "ebayui-core-star-rating",
    "tags-dir": "./icons"
}
