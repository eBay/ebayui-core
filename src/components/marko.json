{
  "<ebay-tooltip-base>": {
    "renderer": "./components/ebay-tooltip-base/index.js"
  },
  "<ebay-tooltip-overlay>": {
    "renderer": "./components/ebay-tooltip-overlay/index.js"
  },
  "<*>": {
    "transformer": "../common/transformers/ebayui-attribute/index.js"
  }
}