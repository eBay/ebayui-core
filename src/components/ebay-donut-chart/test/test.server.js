import { use } from "chai";
import { composeStories } from "@storybook/marko";
import { snapshotHTML } from "../../../common/test-utils/snapshots";
import * as stories from "../donut-chart.stories";

const { Standard, TwoValues, FiveValues, NoMetrics } = composeStories(stories);
const htmlSnap = snapshotHTML(__dirname);

use(require("chai-dom"));

describe("section-title", () => {
    it("renders defaults", async () => {
        await htmlSnap(Standard);
    });

    it("renders with two values", async () => {
        await htmlSnap(TwoValues);
    });

    it("renders with five values", async () => {
        await htmlSnap(FiveValues);
    });

    it("renders with no metrics", async () => {
        await htmlSnap(NoMetrics);
    });
});
