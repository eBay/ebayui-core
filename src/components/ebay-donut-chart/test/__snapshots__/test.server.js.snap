// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`section-title > renders defaults 1`] = `
"[36m<DocumentFragment>[39m
  [36m<div[39m
    [33mclass[39m=[32m"donut-chart"[39m
  [36m>[39m
    [36m<div[39m
      [33mclass[39m=[32m"donut-chart__grid"[39m
    [36m>[39m
      [36m<div[39m
        [33mclass[39m=[32m"donut-chart__title"[39m
      [36m>[39m
        [0mDonut chart title[0m
      [36m</div>[39m
      [36m<div[39m
        [33mclass[39m=[32m"donut-chart__metric"[39m
      [36m>[39m
        [36m<div[39m
          [33mclass[39m=[32m"donut-chart__metric-value"[39m
        [36m>[39m
          [0m174 Total[0m
        [36m</div>[39m
        [36m<div[39m
          [33mclass[39m=[32m"donut-chart__metric-label"[39m
        [36m>[39m
          [0m3.78% return rate[0m
        [36m</div>[39m
      [36m</div>[39m
      [36m<div[39m
        [33mclass[39m=[32m"donut-chart__graph"[39m
        [33mdata-marko-key[39m=[32m"9 s0"[39m
        [33mid[39m=[32m"ebay-donut-graph-s0"[39m
      [36m/>[39m
      [36m<div[39m
        [33mclass[39m=[32m"donut-chart__legend"[39m
      [36m>[39m
        [36m<div[39m
          [33mclass[39m=[32m"chart-legend"[39m
        [36m>[39m
          [36m<dl[39m
            [33mclass[39m=[32m"chart-legend__list"[39m
          [36m>[39m
            [36m<div[39m
              [33mclass[39m=[32m"chart-legend__item"[39m
            [36m>[39m
              [36m<dt[39m
                [33mclass[39m=[32m"chart-legend__label"[39m
              [36m>[39m
                [0mPortion 1[0m
              [36m</dt>[39m
              [36m<dd[39m
                [33mclass[39m=[32m"chart-legend__value"[39m
              [36m>[39m
                [0m20[0m
              [36m</dd>[39m
            [36m</div>[39m
            [36m<div[39m
              [33mclass[39m=[32m"chart-legend__item"[39m
            [36m>[39m
              [36m<dt[39m
                [33mclass[39m=[32m"chart-legend__label"[39m
              [36m>[39m
                [0mPortion 2[0m
              [36m</dt>[39m
              [36m<dd[39m
                [33mclass[39m=[32m"chart-legend__value"[39m
              [36m>[39m
                [0m40[0m
              [36m</dd>[39m
            [36m</div>[39m
            [36m<div[39m
              [33mclass[39m=[32m"chart-legend__item"[39m
            [36m>[39m
              [36m<dt[39m
                [33mclass[39m=[32m"chart-legend__label"[39m
              [36m>[39m
                [0mPortion 3[0m
              [36m</dt>[39m
              [36m<dd[39m
                [33mclass[39m=[32m"chart-legend__value"[39m
              [36m>[39m
                [0m50[0m
              [36m</dd>[39m
            [36m</div>[39m
          [36m</dl>[39m
        [36m</div>[39m
      [36m</div>[39m
    [36m</div>[39m
  [36m</div>[39m
[36m</DocumentFragment>[39m"
`;

exports[`section-title > renders with five values 1`] = `
"[36m<DocumentFragment>[39m
  [36m<div[39m
    [33mclass[39m=[32m"donut-chart"[39m
  [36m>[39m
    [36m<div[39m
      [33mclass[39m=[32m"donut-chart__grid"[39m
    [36m>[39m
      [36m<div[39m
        [33mclass[39m=[32m"donut-chart__title"[39m
      [36m>[39m
        [0mDonut chart with five values[0m
      [36m</div>[39m
      [36m<div[39m
        [33mclass[39m=[32m"donut-chart__metric"[39m
      [36m>[39m
        [36m<div[39m
          [33mclass[39m=[32m"donut-chart__metric-value"[39m
        [36m>[39m
          [0m174 Total[0m
        [36m</div>[39m
        [36m<div[39m
          [33mclass[39m=[32m"donut-chart__metric-label"[39m
        [36m>[39m
          [0m3.78% return rate[0m
        [36m</div>[39m
      [36m</div>[39m
      [36m<div[39m
        [33mclass[39m=[32m"donut-chart__graph"[39m
        [33mdata-marko-key[39m=[32m"9 s0"[39m
        [33mid[39m=[32m"ebay-donut-graph-s0"[39m
      [36m/>[39m
      [36m<div[39m
        [33mclass[39m=[32m"donut-chart__legend"[39m
      [36m>[39m
        [36m<div[39m
          [33mclass[39m=[32m"chart-legend"[39m
        [36m>[39m
          [36m<dl[39m
            [33mclass[39m=[32m"chart-legend__list"[39m
          [36m>[39m
            [36m<div[39m
              [33mclass[39m=[32m"chart-legend__item"[39m
            [36m>[39m
              [36m<dt[39m
                [33mclass[39m=[32m"chart-legend__label"[39m
              [36m>[39m
                [0mPortion 1[0m
              [36m</dt>[39m
              [36m<dd[39m
                [33mclass[39m=[32m"chart-legend__value"[39m
              [36m>[39m
                [0m20[0m
              [36m</dd>[39m
            [36m</div>[39m
            [36m<div[39m
              [33mclass[39m=[32m"chart-legend__item"[39m
            [36m>[39m
              [36m<dt[39m
                [33mclass[39m=[32m"chart-legend__label"[39m
              [36m>[39m
                [0mPortion 2[0m
              [36m</dt>[39m
              [36m<dd[39m
                [33mclass[39m=[32m"chart-legend__value"[39m
              [36m>[39m
                [0m40[0m
              [36m</dd>[39m
            [36m</div>[39m
            [36m<div[39m
              [33mclass[39m=[32m"chart-legend__item"[39m
            [36m>[39m
              [36m<dt[39m
                [33mclass[39m=[32m"chart-legend__label"[39m
              [36m>[39m
                [0mPortion 3[0m
              [36m</dt>[39m
              [36m<dd[39m
                [33mclass[39m=[32m"chart-legend__value"[39m
              [36m>[39m
                [0m50[0m
              [36m</dd>[39m
            [36m</div>[39m
            [36m<div[39m
              [33mclass[39m=[32m"chart-legend__item"[39m
            [36m>[39m
              [36m<dt[39m
                [33mclass[39m=[32m"chart-legend__label"[39m
              [36m>[39m
                [0mPortion 4[0m
              [36m</dt>[39m
              [36m<dd[39m
                [33mclass[39m=[32m"chart-legend__value"[39m
              [36m>[39m
                [0m30[0m
              [36m</dd>[39m
            [36m</div>[39m
            [36m<div[39m
              [33mclass[39m=[32m"chart-legend__item"[39m
            [36m>[39m
              [36m<dt[39m
                [33mclass[39m=[32m"chart-legend__label"[39m
              [36m>[39m
                [0mPortion 5[0m
              [36m</dt>[39m
              [36m<dd[39m
                [33mclass[39m=[32m"chart-legend__value"[39m
              [36m>[39m
                [0m10[0m
              [36m</dd>[39m
            [36m</div>[39m
          [36m</dl>[39m
        [36m</div>[39m
      [36m</div>[39m
    [36m</div>[39m
  [36m</div>[39m
[36m</DocumentFragment>[39m"
`;

exports[`section-title > renders with no metrics 1`] = `
"[36m<DocumentFragment>[39m
  [36m<div[39m
    [33mclass[39m=[32m"donut-chart"[39m
  [36m>[39m
    [36m<div[39m
      [33mclass[39m=[32m"donut-chart__grid"[39m
    [36m>[39m
      [36m<div[39m
        [33mclass[39m=[32m"donut-chart__title"[39m
      [36m>[39m
        [0mDonut chart title, no metrics[0m
      [36m</div>[39m
      [36m<div[39m
        [33mclass[39m=[32m"donut-chart__graph"[39m
        [33mdata-marko-key[39m=[32m"9 s0"[39m
        [33mid[39m=[32m"ebay-donut-graph-s0"[39m
      [36m/>[39m
      [36m<div[39m
        [33mclass[39m=[32m"donut-chart__legend"[39m
      [36m>[39m
        [36m<div[39m
          [33mclass[39m=[32m"chart-legend"[39m
        [36m>[39m
          [36m<dl[39m
            [33mclass[39m=[32m"chart-legend__list"[39m
          [36m>[39m
            [36m<div[39m
              [33mclass[39m=[32m"chart-legend__item"[39m
            [36m>[39m
              [36m<dt[39m
                [33mclass[39m=[32m"chart-legend__label"[39m
              [36m>[39m
                [0mPortion 1[0m
              [36m</dt>[39m
              [36m<dd[39m
                [33mclass[39m=[32m"chart-legend__value"[39m
              [36m>[39m
                [0m20[0m
              [36m</dd>[39m
            [36m</div>[39m
            [36m<div[39m
              [33mclass[39m=[32m"chart-legend__item"[39m
            [36m>[39m
              [36m<dt[39m
                [33mclass[39m=[32m"chart-legend__label"[39m
              [36m>[39m
                [0mPortion 2[0m
              [36m</dt>[39m
              [36m<dd[39m
                [33mclass[39m=[32m"chart-legend__value"[39m
              [36m>[39m
                [0m40[0m
              [36m</dd>[39m
            [36m</div>[39m
            [36m<div[39m
              [33mclass[39m=[32m"chart-legend__item"[39m
            [36m>[39m
              [36m<dt[39m
                [33mclass[39m=[32m"chart-legend__label"[39m
              [36m>[39m
                [0mPortion 3[0m
              [36m</dt>[39m
              [36m<dd[39m
                [33mclass[39m=[32m"chart-legend__value"[39m
              [36m>[39m
                [0m50[0m
              [36m</dd>[39m
            [36m</div>[39m
            [36m<div[39m
              [33mclass[39m=[32m"chart-legend__item"[39m
            [36m>[39m
              [36m<dt[39m
                [33mclass[39m=[32m"chart-legend__label"[39m
              [36m>[39m
                [0mPortion 4[0m
              [36m</dt>[39m
              [36m<dd[39m
                [33mclass[39m=[32m"chart-legend__value"[39m
              [36m>[39m
                [0m30[0m
              [36m</dd>[39m
            [36m</div>[39m
          [36m</dl>[39m
        [36m</div>[39m
      [36m</div>[39m
    [36m</div>[39m
  [36m</div>[39m
[36m</DocumentFragment>[39m"
`;

exports[`section-title > renders with two values 1`] = `
"[36m<DocumentFragment>[39m
  [36m<div[39m
    [33mclass[39m=[32m"donut-chart"[39m
  [36m>[39m
    [36m<div[39m
      [33mclass[39m=[32m"donut-chart__grid"[39m
    [36m>[39m
      [36m<div[39m
        [33mclass[39m=[32m"donut-chart__title"[39m
      [36m>[39m
        [0mDonut chart with two values[0m
      [36m</div>[39m
      [36m<div[39m
        [33mclass[39m=[32m"donut-chart__metric"[39m
      [36m>[39m
        [36m<div[39m
          [33mclass[39m=[32m"donut-chart__metric-value"[39m
        [36m>[39m
          [0m174 Total[0m
        [36m</div>[39m
        [36m<div[39m
          [33mclass[39m=[32m"donut-chart__metric-label"[39m
        [36m>[39m
          [0m3.78% return rate[0m
        [36m</div>[39m
      [36m</div>[39m
      [36m<div[39m
        [33mclass[39m=[32m"donut-chart__graph"[39m
        [33mdata-marko-key[39m=[32m"9 s0"[39m
        [33mid[39m=[32m"ebay-donut-graph-s0"[39m
      [36m/>[39m
      [36m<div[39m
        [33mclass[39m=[32m"donut-chart__legend"[39m
      [36m>[39m
        [36m<div[39m
          [33mclass[39m=[32m"chart-legend"[39m
        [36m>[39m
          [36m<dl[39m
            [33mclass[39m=[32m"chart-legend__list"[39m
          [36m>[39m
            [36m<div[39m
              [33mclass[39m=[32m"chart-legend__item"[39m
            [36m>[39m
              [36m<dt[39m
                [33mclass[39m=[32m"chart-legend__label"[39m
              [36m>[39m
                [0mPortion 1[0m
              [36m</dt>[39m
              [36m<dd[39m
                [33mclass[39m=[32m"chart-legend__value"[39m
              [36m>[39m
                [0m20[0m
              [36m</dd>[39m
            [36m</div>[39m
            [36m<div[39m
              [33mclass[39m=[32m"chart-legend__item"[39m
            [36m>[39m
              [36m<dt[39m
                [33mclass[39m=[32m"chart-legend__label"[39m
              [36m>[39m
                [0mPortion 2[0m
              [36m</dt>[39m
              [36m<dd[39m
                [33mclass[39m=[32m"chart-legend__value"[39m
              [36m>[39m
                [0m40[0m
              [36m</dd>[39m
            [36m</div>[39m
          [36m</dl>[39m
        [36m</div>[39m
      [36m</div>[39m
    [36m</div>[39m
  [36m</div>[39m
[36m</DocumentFragment>[39m"
`;
